### PR TITLE
Feature/app 6286 sync cmp bkt creditors to ods

### PR DESCRIPTION
**Pull request by john5macd**
Two main things here:

1. Code to check the configured view has the required fields.  They provider can't change things underneath us.
2. Refactor target to introduce Repository pattern to allow unit test.  In first instance I tried:

- Using InMemory Database - though that then required reaching into the database to pull out values for assert - not much fun, and In Memory databases not recommended where transactions are concerned - which we have
- Mocking entity framework core - managed to get an online example partially working - though example did not handle all latest feature of EF Core, needed to support current calls, to support believe would have need to upgrade to .Net 10, in end felt like was chasing unicorns so opted for Repository pattern based on online recommendation from a Microsoft bod

### CODE REVIEW

File: INSS.FIP.Data/CMPDataSource/sourceCMPDbContextExtension.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace INSS.FIP.Data.CMPDataSource
+{
+    public static class sourceCMPDbContextExtension
```
Comment (VasInsolvencyService) [inline INSS.FIP.Data/CMPDataSource/sourceCMPDbContextExtension.cs:10]: PascalCase for class names

File: INSS.FIP.Functions/Program.cs [hunk 2]
```diff
@@ -36,12 +38,14 @@
             return new sourceCMPDbContext(connectionString, configuration);
         });
 
-        services.AddTransient<targetCMPDbContext>(sp =>
+        services.AddScoped<targetCMPDbContext>(sp =>
```
Comment (VasInsolvencyService) [inline INSS.FIP.Functions/Program.cs:41]: whats the reason to move to AddScoped

File: INSS.FIP.Functions.UnitTests/FunctionsTests/CMPSyncTests/BankruptcyCreditorsDataSourceProviderTests.cs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+            //Act
+            await aProvider.TruncateAndInsertAsync(new List<CentrallyManagedPartyModel> 
+            {
+                new CentrallyManagedPartyModel (),
+                new CentrallyManagedPartyModel ()
+            });
+
+            //Assert 
+            Assert.Equal(entitiesToAdd, args, new BankruptcyCreditorsListListComparer());
```
Comment (VasInsolvencyService) [inline INSS.FIP.Functions.UnitTests/FunctionsTests/CMPSyncTests/BankruptcyCreditorsDataSourceProviderTests.cs:58]: Instead of comparing entire list , shall we do 
Assert.NotNull(args);
Assert.Equal(2, args.Count);

so its more readable

**Review by VasInsolvencyService — COMMENTED** [review]
I have few queries relate to  BankruptcyCreditorsDataSourceProviderTests.cs‎ , i need to chat with you before adding comments.

File: INSS.FIP.Functions/Program.cs (hunk 2 above)
Comment (john5macd) [reply INSS.FIP.Functions/Program.cs:41]: This comes back to recommendation when using Repository pattern / transactions.  As I understand it with scoped each "request" ends up guaranteed to get the same db context - probably a mute point for us as pretty much doing the same read/write from the one method.  Though we do have the BankruptciesCreditorsProvider calling context 4 times and wanted to make sure separate calls were using the same context - not to important now but potentially in the future we may have DROs & BKTs being updated from separate methods where the same context may be important.

File: INSS.FIP.Data/CMPDataSource/sourceCMPDbContextExtension.cs (hunk 1 above)
Comment (john5macd) [reply INSS.FIP.Data/CMPDataSource/sourceCMPDbContextExtension.cs:10]: I'll look to change in an additional PR - after the next... otherwise things will get even more confusing.

File: INSS.FIP.Functions.UnitTests/FunctionsTests/CMPSyncTests/BankruptcyCreditorsDataSourceProviderTests.cs (hunk 3 above)
Comment (john5macd) [reply INSS.FIP.Functions.UnitTests/FunctionsTests/CMPSyncTests/BankruptcyCreditorsDataSourceProviderTests.cs:58]: The comparing of lists helps compare the total input / output of the BankruptcyCreditorsProvider. BankruptcyCreditorsProvider (& BankruptcyCreditorsRepository) effectively form an Infrastructure layer in terms of clean architecture pattern - the two required are required so we can effectively unit test. Now the folder structure doesn't reflect a clean architecture pattern because we've started from a point of existing FIP, and I don't think we should be refactoring FIP at this stage so it screams clean architecture.  I have put much logic into BankruptcyCreditorsProvider, rather that the Application Layer (which in this case is => INSS.FIP.Functions\Helper\DbSyncCMPData.cs), because that logic is specific to Bankruptcy Creditors... we need to sort, convert nulls to spaces (for n fields within the data), add an identifier, and drop a few duplicates.  This way we keep the application layer to get data from source, then save to target.  If we don't compare the lists then potentially have to break the unit test into much smaller units = more code, one test for each property = let's say 8 different tests, we would also then have to restructure code to say test the numbering, and dropping of the duplicate records.  This way we can do it en masse - you will see in next pull request, I have broken this down to separate tests for specific requirements, also the way I've handled converting null => "" is handled on the class properties and covered by their own unit tests... so covered twice.

**Review by VasInsolvencyService — APPROVED** [review]
